atr/get/sbom.py

```diff
@@ -468,58 +547,86 @@ def _vulnerability_scan_find_in_progress_task(
+                    _cdx_to_osv(v)
+                    for v in vulns
+                    if (v.affects is not None) and (component in [a.get("ref") for a in v.affects])
+                ],
+            ),
+            previous_vulns,
+        )
+        total_new = total_new + new
+
+    new_str = f" ({total_new!s} new since last release)" if (total_new > 0) else ""
+    block.p[f"Vulnerabilities{new_str} found in {len(components)} components:"]
+    block.append(new_block)
 
-        block.p[f"Scan found vulnerabilities in {len(components)} components:"]
 
-        for component in components:
-            _vulnerability_component_details_osv(block, component)
+def _vulnerability_results_from_scan(task: sql.Task, block: htm.Block, previous_vulns: dict[str, str] | None) -> None:
+    total_new = 0
+    new_block = htm.Block()
+    task_result = task.result
+    if not isinstance(task_result, results.SBOMOSVScan):
+        block.p["Invalid scan result format."]
+        return
 
+    components = task_result.components
+    ignored = task_result.ignored
+    ignored_count = len(ignored)
+
+    if not components:
+        block.p["No vulnerabilities found."]
         if ignored_count > 0:
             component_word = "component was" if (ignored_count == 1) else "components were"
             block.p[f"{ignored_count} {component_word} ignored due to missing PURL or version information:"]
             block.p[f"{','.join(ignored)}"]
-    else:
-        if len(vulns) == 0:
-            block.p["No vulnerabilities listed in this SBOM."]
-            return
-        components = {a.get("ref", "") for v in vulns if v.affects is not None for a in v.affects}
+        return
 
-        if len(scans) > 0:
-            block.p["This SBOM was scanned for vulnerabilities at revision ", htm.code[scans[-1]], "."]
+    for component in components:
+        new = _vulnerability_component_details_osv(new_block, component, previous_vulns)
+        total_new = total_new + new
 
-        block.p[f"Vulnerabilities found in {len(components)} components:"]
+    new_str = f" ({total_new!s} new since last release)" if (total_new > 0) else ""
+    block.p[f"Scan found vulnerabilities{new_str} in {len(components)} components:"]
 
-        for component in components:
-            _vulnerability_component_details_osv(
-                block,
-                results.OSVComponent(
-                    purl=component,
-                    vulnerabilities=[
-                        _cdx_to_osv(v)
-                        for v in vulns
-                        if (v.affects is not None) and (component in [a.get("ref") for a in v.affects])
-                    ],
-                ),
-            )
+    if ignored_count > 0:
+        component_word = "component was" if (ignored_count == 1) else "components were"
+        block.p[f"{ignored_count} {component_word} ignored due to missing PURL or version information:"]
+        block.p[f"{','.join(ignored)}"]
```

Comment:
I think this is missing a `block.append(new_block)` after it, to make it work the same as `_vulnerability_results_from_bom` which does do that.